### DISCUSSION TITLE
ueransim how to use iperf3 to send udp packet

### DISCUSSION THREAD

**Discussion by zh7606559342012** [aligungr/UERANSIM#474] [Q&A]
ueransim how to use iperf3 to send udp packet

**Comment by aligungr** [comment]
@zh7606559342012 

You can try the following:

---

In UEs machine, run the following:

`./nr-binder {PDU-IP-ADDRESS} iperf3 -c {IP-OF-CORE-NETWORK-MACHINE} -i 1 -t 2000`

for example

`./nr-binder 10.45.0.2 iperf3 -c 192.168.1.40 -i 1 -t 2000`

---

In core network machine, run the following:

`iperf3 -s`

**Reply by elrandira** [reply]
interesting, in our case we use this directly without nr-binder:
```iperf3 -c {IP-OF-CORE-NETWORK-MACHINE} -i 1 -t 2000 -B ${PDU-IP-ADDRESS}```

**Reply by bintangcahya696** [reply]
Hi,

I'm currently testing UDP traffic between UERANSIM and Free5GC using iperf3 and nr-binder. While TCP tests work fine, I'm having trouble getting UDP to work properly.

Here’s the setup: sudo ./nr-binder 10.60.0.1 iperf3 -c 192.168.56.101 -i  1 -t  60 -b 10M -u
iperf3: error - unable to read from stream socket: Resource temporarily unavailaibe

**Comment by bintangcahya696** [comment]
Hi,

I'm currently testing UDP traffic between UERANSIM and Free5GC using iperf3 and nr-binder. While TCP tests work fine, I'm having trouble getting UDP to work properly.

Here’s the setup: sudo ./nr-binder 10.60.0.1 iperf3 -c 192.168.56.101 -i 1 -t 60 -b 10M -u
iperf3: error - unable to read from stream socket: Resource temporarily unavailaibe
